keras/engine/topology.py

```diff
@@ -1317,13 +1318,34 @@ def get_output_shape_for(self, input_shape):
         elif self.mode == 'cos':
             return (input_shapes[0][0], 1)
 
-    def compute_mask(self, input, mask=None):
-        '''TODO: add mask merging support
-        '''
-        if mask is not None and any(mask):
-            raise Exception('Merge does not support masking, ' +
-                            'but was passed an input mask: ' + str(mask))
-        return None
+    def compute_mask(self, inputs, mask=None):
+        if mask is None or not any([m is not None for m in mask]):
+            return None
+
+        assert hasattr(mask, '__len__') and len(mask) == len(inputs)
+
+        if self.mode in ['sum', 'mul', 'ave']:
+            bool_type = 'bool' if K._BACKEND == 'tensorflow' else 'int32'
```

Comment:
The Tensorflow backend does not support logical AND operations for int types, and Theano does not have a boolean type.